utils/zerovec/src/ule/option.rs

```diff
@@ -47,28 +49,30 @@ impl<T: Copy> OptionULE<T> {
 
 // Safety (based on the safety checklist on the ULE trait):
 //  1. OptionULE does not include any uninitialized or padding bytes.
-//     (achieved by `#[repr(packed)]` on a struct containing only ULE fields)
+//     (achieved by `#[repr(packed)]` on a struct containing only ULE fields,
+//     in the context of this impl. The MaybeUninit is valid for all byte sequences)
 //  2. OptionULE is aligned to 1 byte.
-//     (achieved by `#[repr(packed)]` on a struct containing only ULE fields)
+//     (achieved by `#[repr(packed)]` on a struct containing only ULE fields, in the context of this impl)
 //  3. The impl of validate_byte_slice() returns an error if any byte is not valid.
 //  4. The impl of validate_byte_slice() returns an error if there are extra bytes.
 //  5. The other ULE methods use the default impl.
 //  6. OptionULE byte equality is semantic equality by relying on the ULE equality
```

Comment:
The byte equality constraint is one of the reasons we don't want to allow junk bytes.

Reply:
I don't think we are allowing junk bytes here